### PR TITLE
Add tokenizer list

### LINKED ISSUE #790: [DOC] Enumerate Filters and Tokenizers AVailable
### Is your feature request related to a problem?
We're missing a piece of documentation regarding what filters and tokenizers are available on OpenSearch.

### What solution would you like?
_A new documentation page with a table that enumerates the available tokenizers and filters are available_

### Do you have any additional context?
https://forum.opensearch.org/t/built-in-filters-and-tokenizers-available-in-opensearch-from-elasticsearch/10229

### PR DESCRIPTION
**Pull request by kolchfa-aws**
Closes #790 

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _analyzers/tokenizers/index.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+`classic` | - Parses strings into tokens on: <br> &emsp; - punctuation characters that are followed by a white space character <br> &emsp; - hyphens if the term does not contain numbers <br> - Removes punctuation <br>  - Leaves URLs and email addresses as single terms | `Part number PA-35234, single-use product (128.32)` <br>becomes<br> [`Part`, `number`, `PA-35234`, `single`, `use`, `product`, `128.32`]
+`thai` | - Parses Thai text into terms | `สวัสดีและยินดีต` <br>becomes<br> [`สวัสด`, `และ`, `ยินดี`, `ต`] 
+
+### Partial word tokenizers
+
+Partial word tokenizers parse text into words and generate fragments of those words for partial word matching.
+
+Tokenizer | Description | Example
+:--- | :--- | :---
+`ngram`| - Parses strings into words on specified characters (for example, punctuation or white space characters) and generates n-grams of each word | `My repo` <br>becomes<br> [`M`, `My`, `y`, `y `, <code>&nbsp;</code>, <code>&nbsp;r</code>, `r`, `re`, `e`, `ep`, `p`, `po`, `o`] because the default n-gram length is 1--2 characters.
```
Comment (vagimeli) [inline _analyzers/tokenizers/index.md:45]: Are the two `y` intentional?

**Review by vagimeli — APPROVED** [review]
LGTM

File: _analyzers/tokenizers/index.md (hunk 1 above)
Comment (kolchfa-aws) [reply _analyzers/tokenizers/index.md:45]: Yup. One has a space and the other one doesn't.

File: _analyzers/tokenizers/index.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+layout: default
+title: Tokenizers
+nav_order: 60
+has_children: false
+has_toc: false
+---
+
+# Tokenizers
+
+A tokenizer receives a stream of characters and splits the text into individual _tokens_. A token is a term (usually, a word) and metadata about this term. For example, a tokenizer can split text on white space so that the text `Actions speak louder than words.` becomes [`Actions`, `speak`, `louder`, `than`, `words.`].
```
Comment (natebower) [inline _analyzers/tokenizers/index.md:11]: Second sentence: A token "includes"?

File: _analyzers/tokenizers/index.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+`whitespace` | - Parses strings into tokens at white space characters | `It’s fun to contribute a brand-new PR or 2 to OpenSearch!` <br>becomes<br> [`It’s`, `fun`, `to`, `contribute`, `a`,`brand-new`, `PR`, `or`, `2`, `to`, `OpenSearch!`] 
+`uax_url_email` | - Similar to the standard tokenizer <br> - Unlike the standard tokenizer, leaves URLs and email addresses as single terms | `It’s fun to contribute a brand-new PR or 2 to OpenSearch opensearch-project@github.com!` <br>becomes<br> [`It’s`, `fun`, `to`, `contribute`, `a`,`brand`, `new`, `PR`, `or`, `2`, `to`, `OpenSearch`, `opensearch-project@github.com`] 
+`classic` | - Parses strings into tokens on: <br> &emsp; - punctuation characters that are followed by a white space character <br> &emsp; - hyphens if the term does not contain numbers <br> - Removes punctuation <br>  - Leaves URLs and email addresses as single terms | `Part number PA-35234, single-use product (128.32)` <br>becomes<br> [`Part`, `number`, `PA-35234`, `single`, `use`, `product`, `128.32`]
```
Comment (natebower) [inline _analyzers/tokenizers/index.md:36]:
```suggestion
`classic` | - Parses strings into tokens on: <br> &emsp; - Punctuation characters that are followed by a white space character <br> &emsp; - Hyphens if the term does not contain numbers <br> - Removes punctuation <br>  - Leaves URLs and email addresses as single terms | `Part number PA-35234, single-use product (128.32)` <br>becomes<br> [`Part`, `number`, `PA-35234`, `single`, `use`, `product`, `128.32`]
```

File: _analyzers/tokenizers/index.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+Partial word tokenizers parse text into words and generate fragments of those words for partial word matching.
+
+Tokenizer | Description | Example
+:--- | :--- | :---
+`ngram`| - Parses strings into words on specified characters (for example, punctuation or white space characters) and generates n-grams of each word | `My repo` <br>becomes<br> [`M`, `My`, `y`, `y `, <code>&nbsp;</code>, <code>&nbsp;r</code>, `r`, `re`, `e`, `ep`, `p`, `po`, `o`] because the default n-gram length is 1--2 characters. 
+`edge_ngram` | - Parses strings into words on specified characters (for example, punctuation or white space characters) and generates edge n-grams of each word (n-grams that start at the beginning of the word) | `My repo` <br>becomes<br> [`M`, `My`] because the default n-gram length is 1--2 characters. 
+
+### Structured text tokenizers
+
+Structured text tokenizers parse structured text, such as identifiers, email addresses, paths, or zip codes.
```
Comment (natebower) [inline _analyzers/tokenizers/index.md:50]:
```suggestion
Structured text tokenizers parse structured text, such as identifiers, email addresses, paths, or ZIP Codes.
```

File: _analyzers/tokenizers/index.md [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+Tokenizer | Description | Example
+:--- | :--- | :---
+`keyword` | - No-op tokenizer <br> - Outputs the entire string unchanged <br> - Can be combined with token filters like lowercase to normalize terms | `My repo` <br>becomes<br> `My repo`
```
Comment (natebower) [inline _analyzers/tokenizers/index.md:54]: This reads as though "like lowercase" should be surrounded by commas.

File: _analyzers/tokenizers/index.md [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+`pattern` | - Uses a regular expression pattern to parse text into terms on a word separator or to capture matching text as terms <br> - Uses [Java regular expressions](https://docs.oracle.com/javase/8/docs/api/java/util/regex/Pattern.html) | `https://opensearch.org/forum` <br>becomes<br> [`https`, `opensearch`, `org`, `forum`] because by default the tokenizer splits terms at word boundaries (`\W+`). Can be configured with a regex pattern.
+`simple_pattern` | - Uses a regular expression pattern to return matching text as terms <br>  - Uses [Lucene regular expressions](https://lucene.apache.org/core/8_7_0/core/org/apache/lucene/util/automaton/RegExp.html)  <br> - Faster than the `pattern` tokenizer because it uses a subset of the `pattern` tokenizer regular expressions |  Returns an empty array by default. Must be configured with a pattern because the pattern defaults to an empty string.
```
Comment (natebower) [inline _analyzers/tokenizers/index.md:56]: Ideally, the phrases in the table would either all use punctuation or not use punctuation. You have a few entries in the right-hand column that use periods, whereas the rest of the table(s) do not use punctuation. Your choice as to how to make consistent.

**Review by natebower — COMMENTED** [review]
@kolchfa-aws Please see my comments and changes and let me know if you have any questions. Thanks!

File: _analyzers/tokenizers/index.md (hunk 5 above)
Comment (kolchfa-aws) [inline _analyzers/tokenizers/index.md:54]:
```suggestion
`keyword` | - No-op tokenizer <br> - Outputs the entire string unchanged <br> - Can be combined with token filters, like lowercase, to normalize terms | `My repo` <br>becomes<br> `My repo`
```

File: _analyzers/tokenizers/index.md (hunk 2 above)
Comment (kolchfa-aws) [reply _analyzers/tokenizers/index.md:11]: Changed to "consists of"

**Pull request merged**